src/warnings.cpp

```diff
@@ -51,36 +53,38 @@ bool GetfLargeWorkInvalidChainFound()
     return fLargeWorkInvalidChainFound;
 }
 
-std::string GetWarnings(const std::string& strFor)
+std::pair<std::string, int64_t> GetWarnings(const std::string& strFor)
 {
+    std::pair<std::string, int64_t> rpc;
+    std::pair<std::string, int64_t> statusbar;
+    statusbar.second = rpc.second = GetTime();
     int nPriority = 0;
-    std::string strStatusBar;
-    std::string strRPC;
 
     LOCK(cs_warnings);
 
     if (!CLIENT_VERSION_IS_RELEASE)
-        strStatusBar = _("This is a pre-release test build - use at your own risk - do not use for mining or merchant applications");
+        statusbar.first = _("This is a pre-release test build - use at your own risk - do not use for mining or merchant applications");
 
     if (GetBoolArg("-testsafemode", DEFAULT_TESTSAFEMODE))
-        strStatusBar = strRPC = "testsafemode enabled";
+        statusbar.first = rpc.first = "testsafemode enabled";
 
     // Misc warnings like out of disk space and clock is wrong
     if (strMiscWarning != "")
     {
         nPriority = 1000;
-        strStatusBar = strMiscWarning;
+        statusbar.first = strMiscWarning;
+        statusbar.second = timestampWarning;
     }
 
     if (fLargeWorkForkFound)
     {
         nPriority = 2000;
-        strStatusBar = strRPC = _("Warning: The network does not appear to fully agree! Some miners appear to be experiencing issues.");
+        statusbar.first = rpc.first = _("Warning: The network does not appear to fully agree! Some miners appear to be experiencing issues.");
```

Comment:
If both `strMiscWarning` and `fLargeWorkForkFound` are set, then `statusbar.first` will have this warning, but `statusbar.second` will still have the value of `timestampWarning`. That would be confusing. Ideally, we should use the timestamp at which the large-work fork was detected. (Similarly for `fLargeWorkInvalidChainFound`.)

Reply:
```suggestion
        statusbar.first = rpc.first = _("Warning: The network does not appear to fully agree! Some miners appear to be experiencing issues.");
        statusbar.second = timestampWarning;
```